mozilla/network-api
# Fix #20 - Allow uploading images

Allows uploading images for the People app.

To test, first add the following lines to the bottom of the admin.py file:
```python
from networkapi.people.models import Person, Link, Team

admin.site.register(Person)
admin.site.register(Link)
admin.site.register(Team)
```
and run `docker-compose build && docker-compose run web sh -c "python manage.py migrate"`

Then, test the following two cases:

__Local uploads__
- Run the app as is and go to [localhost:5000/admin](localhost:5000/admin)
- Create a Team and Link first
- Create a Person and upload a picture for the `image` input
- Click Save and then open the person you just created. You'll see that there is a link to the image currently used (which you should click on and see if it works)
- Also check that there is an `app/media` directory where the images are stored and that it is not picked up by git.

__S3 uploads__
- Create an S3 bucket, an IAM user (with PutObjectAcl permissions to the bucket) and a CloudFront distribution.
- Fill out the following values in your `env.default` file:
```
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_STORAGE_BUCKET_NAME=
AWS_STORAGE_ROOT=<the root object path in your bucket where all the assets should be uploaded to (if any). Make sure it doesn't begin with a slash>
AWS_S3_CUSTOM_DOMAIN=<cloudfront domain without protocol. Make sure it doesn't end with a slash.>
```
- Also set `USE_S3` in the `env.default` file to `True`
- Rebuild and run your docker container - `docker-compose up --build`
- Go to [localhost:5000/admin](localhost:5000/admin)
- Create a Team and Link first
- Create a Person and upload a picture for the `image` input
- Click Save and then open the person you just created. You'll see that there is a link to the image currently used (which you should click on and see if it works)
